lib/spack/spack/installer.py

```diff
@@ -1730,8 +1737,14 @@ def _push_task(self, task: BuildTask) -> None:
             return
 
         # Remove any associated build task since its sequence will change
+        def ord(num):
+            suffixes = {"1": "st", "2": "nd", "3": "rd"}
+            d = str(num)[-1]
+            suffix = suffixes[d] if d in suffixes and num % 100 not in [11, 12, 13] else "th"
+            return f"{num}{suffix}"
```

Comment:
This probably belongs in llnl.util.string.

Reply:
Done

Reply:
It's now in `llnl.string`.